File: frontend/src/types/book.ts
export interface Book {
  id: number;
  title: string;
  author: string;
  genre: string;
  description?: string;
  copy: number;
  price: number;
}

export interface BookRequest {
  title: string;
  author: string;
  genre: string;
  description?: string;
  copy: number;
  price: number;
}

File: backend/src/main/java/com/bookstore/entity/Book.java
package com.bookstore.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "book")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Book {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 255)
    private String title;

    @Column(nullable = false, length = 255)
    private String author;

    @Column(length = 100)
    private String genre;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false)
    private Integer copy;

    @Column(nullable = false, precision = 10, scale = 2)
    private Double price;
}
